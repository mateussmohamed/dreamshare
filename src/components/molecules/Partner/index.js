import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const PartnerItem = styled.article`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 0 10px;
`

const PartnerHeader = styled.header`
  width: 120px;
  height: 120px;
  position: relative;
`

const PartnerAvatar = styled.div`
  width: 120px;
  height: 120px;
  border-radius: 120px;
  background-color: grey;
  background-image: url(${props => props.src});
  background-repeat: no-repeat;
  background-position: center center;
`

const PartnerIcon = styled.div`
  width: 32px;
  height: 32px;
  border-radius: 32px;
  background-color: #f35e5a;
  border: 4px solid #ffffff;
  position: absolute;
  right: -1px;
  bottom: -1px;
  background-image: url(${props => props.icon});
  background-size: 16px 16px;
  background-position: center center;
  background-repeat: no-repeat;
`

const PartnerName = styled.h3`
  ${props => props.theme.typography.montserratStyle};
  font-size: 14px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  text-align: center;
  margin: 15px 0;
`

const PartnerBio = styled.p`
  ${props => props.theme.typography.robotoStyle};
  font-size: 14px;
  text-align: center;
  margin: 0 0 15px 0;
`

const Partner = ({ name, bio, avatar, icon }) => (
  <PartnerItem>
    <PartnerHeader>
      <PartnerAvatar src={avatar} />
      <PartnerIcon icon={icon} />
    </PartnerHeader>
    <PartnerName>{name}</PartnerName>
    <PartnerBio>{bio}</PartnerBio>
  </PartnerItem>
)

Partner.propTypes = {
  name: PropTypes.string.isRequired,
  bio: PropTypes.string.isRequired,
  avatar: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
}

export default Partner
