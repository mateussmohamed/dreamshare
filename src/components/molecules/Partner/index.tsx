import React from 'react'

import PropTypes from 'prop-types'

import {
  PartnerCard,
  PartnerHeader,
  PartnerAvatar,
  PartnerIcon,
  PartnerName,
  PartnerBio,
} from '../../atoms'

export const Partner = ({ name, bio, avatar, icon }) => (
  <PartnerCard>
    <PartnerHeader>
      <PartnerAvatar src={avatar} />
      <PartnerIcon icon={icon} />
    </PartnerHeader>
    <PartnerName>{name}</PartnerName>
    <PartnerBio>{bio}</PartnerBio>
  </PartnerCard>
)

Partner.propTypes = {
  name: PropTypes.string.isRequired,
  bio: PropTypes.string.isRequired,
  avatar: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
}
