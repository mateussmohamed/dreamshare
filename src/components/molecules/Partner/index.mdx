---
name: Partner
menu: Molecules
---

import { Playground, PropsTable } from 'docz'
import Partner from './index'

# Partner
<PropsTable of={Partner} />

<Playground>
  <Partner
    name="Bradley Hunter"
    avatar="/assets/images/partner-1.jpg"
    bio="Based in Chicago. I love playing tennis and loud music."
    interests="music"
  />
</Playground>
