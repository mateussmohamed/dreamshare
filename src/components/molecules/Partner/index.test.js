import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import Partner from './index'

describe('[<Partner />]', () => {
  const wrap = (props = {}) => shallow(<Partner {...props} />).dive()

  test('matches to snapshot', () => {
    const wrapper = wrap({
      name: 'Bradley Hunter',
      avatar: '/assets/images/partner-1.jpg',
      bio: 'Based in Chicago. I love playing tennis and loud music.',
      interests: 'music',
    })

    expect(wrapper).toMatchSnapshot()
  })
})
