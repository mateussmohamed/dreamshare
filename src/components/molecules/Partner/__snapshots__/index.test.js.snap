// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<Partner />] matches to snapshot 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "Partner__PartnerItem-sc-1mrrjxf-0",
        "isStatic": true,
        "rules": Array [
          "display:flex;flex-direction:column;align-items:center;",
        ],
      },
      "displayName": "Partner__PartnerItem",
      "render": [Function],
      "styledComponentId": "Partner__PartnerItem-sc-1mrrjxf-0",
      "target": "article",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  <Partner__PartnerHeader>
    <Partner__PartnerAvatar
      src="/assets/images/partner-1.jpg"
    />
    <Partner__PartnerIcon
      icon="music"
    />
  </Partner__PartnerHeader>
  <Partner__PartnerName>
    Bradley Hunter
  </Partner__PartnerName>
  <Partner__PartnerBio>
    Based in Chicago. I love playing tennis and loud music.
  </Partner__PartnerBio>
</StyledComponent>
`;
